sharedbuffers/mapped_struct.py

```diff
@@ -966,9 +968,10 @@ def __iter__(self):
 
     def __reversed__(self):
         l = len(self)
-        if l > 0:
-            for i in xrange(l - 1, -1, -1):
-                yield self[i]
+        if l == 0:
+            return self._make_empty()
+        for i in xrange(l - 1, -1, -1):
```

Comment:
Declare `i` as `cython.longlong` to get an optimized iteration.

Same in other places where you do this.